specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,316 @@
+  /// The default settings will also be updated upon navigation to an external
+  /// protocol. 
+  ///
+  /// If the request is made from a trustworthy origin
+  /// (https://w3c.github.io/webappsec-secure-contexts#potentially-trustworthy-origin) 
+  /// a checkmark box will be displayed on the default browser UI that gives the user 
+  /// the option to always allow the external protocol to launch from this origin.
+  /// If the user checks this box, upon the next request from that origin to the 
+  /// protocol, the event will still be raised. 
+  ///
+  /// If the request is initiated by a cross-origin iframe without a user gesture, 
+  /// the request will be blocked and the `LaunchingExternalProtocol` event will not 
+  /// be raised. 
+  /// If the request is initiated by a browser popup window or in another case in 
+  /// which the embedded browser is null, the `LaunchingExternalProtocol` event will 
+  /// not be raised and the request will continue as normal and display the default browser
+  /// UI dialog to the user.
+  /// \snippet SettingsComponent.cpp LaunchingExternalProtocol
+  HRESULT add_LaunchingExternalProtocol(
+      [in] ICoreWebView2LaunchingExternalProtocolEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with
+  /// `add_LaunchingExternalProtocol`.
+  HRESULT remove_LaunchingExternalProtocol(
+      [in] EventRegistrationToken token);
+  }
+
+/// Receives the `LaunchingExternalProtocol` event.
+[uuid(e5fea648-79c9-47aa-8314-f471fe627649), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventHandler: IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2LaunchingExternalProtocolEventArgs* args);
+}
+
+/// Event args for `LaunchingExternalProtocol` event.
+[uuid(fc43b557-9713-4a67-af8d-a76ef3a206e8), object, pointer_default(unique)] 
+interface ICoreWebView2LaunchingExternalProtocolEventArgs: IUnknown {
+  /// The URI of the requested external protocol.
+
+  [propget] HRESULT Uri([out, retval] LPWSTR* uri);
+
+  /// The origin initiating the external protocol launch.
+  /// The origin will be empty if the WebView2 navigates to the external protocol.
+
+  [propget] HRESULT InitiatingOrigin([out, retval] LPWSTR* uri);
+
+  /// `TRUE` when the external protocol request was initiated through a user gesture.
+  ///
+  /// \> [!NOTE]\n\> Being initiated through a user gesture does not mean that user intended
+  /// to access the associated resource.
+
+  [propget] HRESULT IsUserInitiated([out, retval] BOOL* isUserInitiated);
+
+  /// `TRUE` when the external protocol request was initiated via a non-main frame that 
+  /// has a different origin then the owning top-level page.
+
+  [propget] HRESULT IsCrossOriginIframe([out, retval] BOOL* isCrossOriginIframe);

```

Comment:
This is a very specific property. How do we expect the end dev to use this?

Reply:
This was intended for an added security measure for the devs to check if it's a cross origin iframe and allow the launch based on their security discretion.

Reply:
Right. It just looks so very specific of both its cross origin and in an iframe. Is that a special case for security when launching an app protocol? Is that the only thing like that? What recommendation do we have for end developers if this is true?

Reply:
Yeah this was just a special security case we agreed on in the Core workstream that we would still the fire event in this case but let the developer know that it was a cross origin iframe. This was the only security case we were concerned about.  The browser allows the request to continue and displays the dialog to the user allowing them to choose.  I don't think we came to a conclusion about a firm recommendation either way although given the security risks we were leaning towards recommending they block the request.

Reply:
It would be good to add that to the documentation then. Something about launches coming from cross origin iframe may represent a security issue and you may not want to allow them or something